tests/app/notifications/test_notifications_letter_callbacks.py

```diff
@@ -218,6 +218,43 @@ def test_process_letter_callback_calls_process_letter_callback_data_task(
     )
 
 
+def test_process_rejected_letter_callback_calls_process_letter_callback_data_task(
```

Comment:
in the test above this one, there are cases for both `DESPATCHED` and `REJECTED` statuses, and page count is not 0. Does this mean we do different things with rejected letters based on different scenarios?